frontend/javascripts/oxalis/model/sagas/saga_helpers.ts

```diff
@@ -39,4 +46,70 @@ export function* takeEveryUnlessBusy(
   yield* takeEvery(actionDescriptor, sagaBusyWrapper);
 }
 
+type EnsureMappingIsPinnedReturnType = {
+  isMappingPinnedIfNeeded: boolean;
+  reason?: string;
+};
+
+export function askUserForPinningActiveMapping(
+  volumeTracing: VolumeTracing,
+  activeMappingByLayer: Record<string, ActiveMappingInfo>,
+): Promise<EnsureMappingIsPinnedReturnType> {
+  return new Promise((resolve, reject) => {
+    if (!volumeTracing.mappingIsPinned) {
+      const pinMapping = async () => {
+        // A mapping that is active and is being annotated needs to be pinned to ensure a consistent state in the future.
+        // See https://github.com/scalableminds/webknossos/issues/5431 for more information.
+        const activeMapping = activeMappingByLayer[volumeTracing.tracingId];
+        if (activeMapping.mappingName) {
+          Store.dispatch(setMappingIsPinnedAction());
+          const message = messages["tracing.pin_mapping_confirmed"](activeMapping.mappingName);
+          Toast.info(message, { timeout: 10000 });
+          console.log(message);
+          resolve({ isMappingPinnedIfNeeded: true, reason: "User confirmed." });
+        } else {
+          // Having an active mapping without a name should be impossible. Therefore, no further error handling is done.
+          reject({ isMappingPinnedIfNeeded: false, reason: "No mapping name." });
+        }
+      };
+      Modal.confirm({
+        title: "Should the active Mapping be pinned?",
+        content: messages["tracing.pin_mapping_info"],
+        okText: "Enable Mapping Permanently",
+        cancelText: "Abort Annotation Action",
+        width: 600,
+        onOk: pinMapping,
+        onCancel: async () => {
+          reject({ isMappingPinnedIfNeeded: false, reason: "User aborted." });
+        },
+      });
+    } else {
+      resolve({ isMappingPinnedIfNeeded: true, reason: "No active mapping." });
```

Comment:
If I'm reading the code correctly, the reason here is rather that the mapping is already pinned, isn't it?

```suggestion
      resolve({ isMappingPinnedIfNeeded: true, reason: "Mapping is already pinned." });
```

However, this is already checked in `ensureMaybeActiveMappingIsPinned` so maybe the whole condition can be removed?